tests/sparseml/core/lifecycle/test_session.py

```diff
@@ -0,0 +1,279 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import defaultdict
+from types import SimpleNamespace
+
+import pytest
+
+from sparseml.core import Framework
+from sparseml.core.event import Event, EventType
+from sparseml.core.lifecycle.event import CallbacksEventLifecycle
+from sparseml.core.lifecycle.session import SparsificationLifecycle
+from sparseml.core.modifier.base import ModifierInterface
+from sparseml.core.state import State
+
+
+class ModifierMock(ModifierInterface):
```

Comment:
Nice, I really like this concept for testing event callbacks

Reply:
Yay!